alacritty/src/display/window.rs

```diff
@@ -145,13 +143,25 @@ impl Window {
             &config.window,
             #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
             x11_visual,
+            #[cfg(target_os = "macos")]
+            tabbing_id,
         );
 
         if let Some(position) = config.window.position {
             window_builder = window_builder
                 .with_position(PhysicalPosition::<i32>::from((position.x, position.y)));
         }
 
+        #[cfg(not(any(target_os = "macos", windows)))]
+        if let Some(token) = event_loop.read_token_from_env() {
+            log::debug!("Activating window with token: {token:?}");
+            window_builder = window_builder.with_activation_token(token);
+            // Remove the token from the env.
+            unsafe {
+                startup_notify::reset_activation_token_env();
+            }
```

Comment:
You'd be surprised, but we already do that. It's unsafe because it touches env variables, as in mutates the global state. You can't do that transparently, because it's again, global variables and it's all racy.

For example if you try to create a window and spawn the process at the same time, winit could implicitly remove the variables, and your process won't activate, so that's why it's all explicit.

Reply:
> You'd be surprised, but we already do that. It's unsafe because it touches env variables, as in mutates the global state. You can't do that transparently, because it's again, global variables and it's all racy.

You mean without using `std::env`? Also which env is this removed from? If it's Alacritty, can't we do this in Rust?

Reply:
I mean, the function simply calls `std::env::remove_var` that's all it does and that's the reason it's unsafe, because it removes the `env` which you could use for launching, so you should take care. Nothing will really break or you can't lead to undefined state, it's just global state mutation basically, which is fine in application code, but I think is unsafe in library code, because it could remove a state set by application.

Reply:
> I mean, the function simply calls std::env::remove_var that's all it does and that's the reason it's unsafe, because it removes the env which you could use for launching, so you should take care.

How is that unsafe?

Reply:
mutates a global state, though, you mutate a global state.

Reply:
Mutating global state that is behind a lock is not unsafe…

Reply:
The issue is that it's not like the mutation itself is unsafe, it's just that if you spawn a process using those variables from thread `A` and then try to remove them on thread `B`, you could simply remove them for `A` and it won't start up notify anything, because the tokens were simply removed.

So the extra care should be taken and set and extra synchronization on top of that should be used.

Reply:
Like it won't break anything or result in UB, it just you could have sometimes things launched with the wrong env, if you don't add extra `Mutex` to sync threads here, thus it was decided to have it unsafe in winit.

Reply:
Yeah that's not how the unsafe keyword is supposed to be used.

Reply:
```
The unsafe keyword has two uses:

    to declare the existence of contracts the compiler can’t check (unsafe fn and unsafe trait),
    and to declare that a programmer has checked that these contracts have been upheld (unsafe {} and unsafe impl, but also unsafe fn – see below).
```

So it clearly a contract which compiler can't check.

Reply:
There are no safety-relevant contracts that could be violated here.